File: L-63.c
#include<stdio.h>
#include<string.h>
#pragma warning(disable:4996)


int main()
{
	int num;
	scanf("%d" ,&num);
	for (size_t j = 0; j < num; j++)
	{
		int s, h, g;
		scanf("%d %d %d",&s,&h,&g);

		if (s==0)
		{
			if (h==129)
			{
				printf("wan mei!");
			}
			else if (h>129)
			{
				printf("ni li hai!");
			}
			else
			{
				printf("duo chi yu!");
			}
			if (g == 25)
			{
				printf(" wan mei!\n");
			}
			else if (g > 25)
			{
				printf(" shao chi rou!\n");
			}
			else
			{
				printf(" duo chi rou!\n");
			}
		}
		else
		{
			if (h == 130)
			{
				printf("wan mei!");
			}
			else if (h > 130)
			{
				printf("ni li hai!");
			}
			else
			{
				printf("duo chi yu!");
			}
			if (g == 27)
			{
				printf(" wan mei!\n");
			}
			else if (g > 27)
			{
				printf(" shao chi rou!\n");
			}
			else
			{
				printf(" duo chi rou!\n");
			}
		}
	}
	return 0;
}

File: L-63.py
num=int(input())

for i in range(num):
    li=input().split()
    for x in range(3):
        li[x]=int(li[x])
    
    if li[0]==1:
        if li[1]>130:
            print('ni li hai!',end=' ')
        elif li[1]<130:
            print('duo chi yu!',end=' ')
        else:
            print('wan mei!',end=' ')
            
        if li[2]>27:
            print('shao chi rou!')
        elif li[2]<27:
            print('duo chi rou!')
        else:
            print('wan mei!')
    else:
        if li[1]>129:
            print('ni li hai!',end=' ')
        elif li[1]<129:
            print('duo chi yu!',end=' ')
        else:
            print('wan mei!',end=' ')
            
        if li[2]>25:
            print('shao chi rou!')
        elif li[2]<25:
            print('duo chi rou!')
        else:
            print('wan mei!')